sed/binning/numba_bin.py

```diff
@@ -177,68 +184,67 @@ def numba_histogramdd(
-        N, D = sample.shape
-
-    if not isinstance(bins, (tuple, list)):
-        bins = D * [bins]
-    Db = len(bins)
-    if isinstance(bins[0], (int, np.int_)):
-        method = "int"
-    elif isinstance(bins[0], np.ndarray):
-        method = "array"
-    else:
-        raise AttributeError(
-            f"bins must be int, np.ndarray or a sequence of the two. "
-            f"Found {type(bins[0])} instead",
-        )
+        num_rows, num_cols = sample.shape  # pylint: disable=unused-variable
+
+    if isinstance(bins, (int, np.int_)):  # bins provided as a single number
+        bins = num_cols * [bins]
+    num_bins = len(bins)  # Number of dimensions in bins
 
-    if Db != D:  # check number of dimensions
+    if num_bins != num_cols:  # check number of dimensions
         raise ValueError(
             "The dimension of bins must be equal to the dimension of the sample x.",
         )
 
-    if method == "array":
+    if not isinstance(bins[0], (int, np.int_, np.ndarray)):
+        raise TypeError(
+            f"bins must be int, np.ndarray or a sequence of the two. "
+            f"Found {type(bins[0])} instead",
+        )
+
+    # method == "array"
+    if isinstance(bins[0], np.ndarray):
+        bins = cast(List[np.ndarray], list(bins))
         hist = _hist_from_bins(
             sample,
             tuple(bins),
             tuple(b.size - 1 for b in bins),
         )
         return hist, bins
 
-    elif method == "int":
-        # normalize the range argument
-        if ranges is None:
-            raise ValueError(
-                "must define a value for ranges when bins is"
-                " the number of bins",
-            )
-        #     ranges = (None,) * D
-        if D == 1 and isinstance(ranges[0], (int, float)):
-            ranges = (ranges,)
-        elif len(ranges) != D:
-            raise ValueError(
-                "range argument must have one entry per dimension",
-            )
+    # method == "int"
+    assert isinstance(bins[0], (int, np.int_))
+    # normalize the range argument
+    if ranges is None:
+        raise ValueError(
+            "must define a value for ranges when bins is"
+            " the number of bins",
+        )
+    if num_cols == 1 and isinstance(ranges[0], (int, float)):
+        ranges = (ranges,)
+    elif len(ranges) != num_cols:
+        raise ValueError(
+            "range argument must have one entry per dimension",
+        )
 
-        ranges = np.asarray(ranges)
-        bins = tuple(bins)
+    # ranges = np.asarray(ranges)
+    bins = tuple(bins)
 
-        # Create edge arrays
-        edges = D * [None]
-        nbin = np.empty(D, int)
+    # Create edge arrays
+    edges: List[Any] = []
+    nbin = np.empty(num_cols, int)
 
-        for i in range(D):
-            edges[i] = np.linspace(*ranges[i, :], bins[i] + 1)
+    for i in range(num_cols):
+        edges.append(np.linspace(ranges[i][0], ranges[i][0], bins[i] + 1))
```

Comment:
How deos this work? linspace from and to the same value... The test passes though, and behaves as numpy. Am I not seeing something?